internal/service/cluster/collector/collector_k8s.go

```diff
@@ -0,0 +1,293 @@
+	resourceTypeID := d.makeClusterResourceTypeID(architecture, strconv.FormatInt(cores, 10))
+
+	extensions := map[string]interface{}{
+		"cpu": map[string]string{
+			"cores":        strconv.FormatInt(cores, 10),
+			"architecture": architecture,
+			"model":        agent.Status.Inventory.Cpu.ModelName,
+		},
+		"memory": map[string]string{
+			"GiB": strconv.FormatInt(agent.Status.Inventory.Memory.PhysicalBytes/(1024*1024*1024), 10),
+		},
+		"role": string(agent.Status.Role),
+		"system": map[string]string{
+			"manufacturer": agent.Status.Inventory.SystemVendor.Manufacturer,
+			"product":      agent.Status.Inventory.SystemVendor.ProductName,
+			"serial":       agent.Status.Inventory.SystemVendor.SerialNumber,
+		},
+		// TODO: add more info for disks, nics, etc...
+	}
+
+	return models.ClusterResource{
+		ClusterResourceID:     resourceID,
+		ClusterResourceTypeID: resourceTypeID,
+		Name:                  agent.Spec.Hostname,
+		Description:           agent.Spec.Hostname,
+		Extensions:            &extensions,
+		ArtifactResourceIDs:   nil,         // TODO: need to link this to template?
+		ResourceID:            uuid.UUID{}, // TODO: need to link this to h/w resource
+		ExternalID:            name,
+		DataSourceID:          d.dataSourceID,
+		GenerationID:          d.generationID,
+	}
+}
+
+// makeNodeClusterTypeName builds a descriptive string to represent the node cluster type
+func (d *K8SDataSource) makeNodeClusterTypeName(vendor, version string) string {
+	return fmt.Sprintf("%s-%s", vendor, version)
+}
+
+// makeClusterResourceTypeID builds a UUID value for this resource type based on its name so that it has
+// a consistent value each time it is created.
+func (d *K8SDataSource) makeNodeClusterTypeID(vendor, version string) uuid.UUID {
+	return utils.MakeUUIDFromName(NodeClusterTypeUUIDNamespace, d.cloudID, d.makeNodeClusterTypeName(vendor, version))
+}
+
+// getExtensionsFromLabels converts a label map to an extensions map
+func (d *K8SDataSource) getExtensionsFromLabels(labels map[string]string) map[string]interface{} {
+	result := map[string]interface{}{}
+	for key, value := range labels {
+		result[key] = value
+	}
+	return result
+}
+
+// convertManagedClusterToNodeCluster converts a ManagedCluster to a ManagedCluster object
+func (d *K8SDataSource) convertManagedClusterToNodeCluster(cluster v1.ManagedCluster) (models.NodeCluster, error) {
+	vendor, found := cluster.Labels["vendor"]
+	if !found {
+		return models.NodeCluster{}, fmt.Errorf("no vendor label found on cluster %s", cluster.Name)
+	}
+
+	var version string
+	if vendor == "OpenShift" {
+		version, found = cluster.Labels[openshiftVersionLabelName]
+		if !found {
+			return models.NodeCluster{}, fmt.Errorf("no version label found on cluster %s", cluster.Name)
+		}
+	}
+
+	// For now continue to generate UUID values based on the string values assigned
+	resourceID := utils.MakeUUIDFromName(NodeClusterUUIDNamespace, d.cloudID, cluster.Name)
```

Comment:
why don´t we make the resourceID equal to the clusterID? It will simplify a lot the association.

Reply:
good idea.  I'll switch that.